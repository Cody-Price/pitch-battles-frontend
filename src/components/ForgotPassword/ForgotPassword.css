.forgot-pw-wrapper {
  position: absolute;
  width: 341px;
  height: 431px;
  left: 500px;
  right: 0;
  top: 0;
  margin: auto;
  transition: 0.3s;
  font-family: "chronotype";
}

.forgot-pw-wrapper.true {
  left: 0;
}

.forgot-password-form .forgot-email-input {
  display: block;
  font-family: "chronotype";
  font-size: 1.1rem;
  padding-left: 1.6rem;
  padding-top: 4px;
  width: 200px;
  height: 1rem;
  margin-right: 2rem;
  margin-top: 2rem;
}

.forgot-my-pw-label {
  margin-top: 2rem;
  margin-bottom: 0;
  font-size: 1.2rem;
}

.password-flexbox {
  display: flex;
  justify-content: space-between;
  height: 2rem;
  margin-bottom: 3rem;
  position: relative;
}

.enter-email-for-new-pw {
  font-size: 1.1rem;
  padding-bottom: 1rem;
  height: 1.2rem;
  margin: 2.4rem 0 0 2rem;
}

.send-new-pw-btn {
  border-radius: 0;
}

.send-new-pw-btn,
.backup-from-forgot-pw,
.have-reset-token {
  margin: 0;
  font-family: "chronotype";
  font-size: 1.1rem;
  padding-top: 4px;
  border-radius: 1px;
}

.send-new-pw-btn,
.backup-from-forgot-pw {
  box-shadow: 2px 2px 2px rgba(0, 0, 0, 0.5);
}

.send-new-pw-btn:active,
.backup-from-forgot-pw:active {
  box-shadow: none;
}

.send-new-pw-btn:focus,
.backup-from-forgot-pw:focus {
  outline: none;
}

.backup-from-forgot-pw {
  margin-top: 8rem;
}
.forgot-pw-lock-icon {
  position: absolute;
  background: url(../Landing/images/lock.png);
  width: 13px;
  height: 16px;
  background-size: cover;
  image-rendering: pixelated;
  left: 85px;
  top: 37px;
}

.pw-sent-success,
.pw-sent-failure {
  position: absolute;
  right: 0;
  left: 500px;
  margin: auto;
  top: 156px;
  transition: 0.3s;
  width: 341px;
}

.pw-sent-success.true,
.pw-sent-failure.true {
  left: 0;
}

.have-reset-token {
  margin: 2.5rem auto;
  display: block;
  background: none;
  border: none;
}

.success-message-reset,
.fail-message-reset,
.no-pw-match-reset {
  position: absolute;
  font-family: "chronotype";
  font-size: 1.1rem;
  left: 400px;
  right: 0;
  margin: auto;
  transition: 0.5s;
  width: 300px;
}

.success-message-reset.true,
.fail-message-reset.true,
.no-pw-match-reset.true {
  left: 0;
}

@media (max-width: 700px) {
  .backup-from-forgot-pw {
    position: absolute;
    margin-left: auto;
    margin-right: auto;
    left: 0;
    right: 0;
    bottom: 2rem;
    width: 4rem;
    padding: 4px 0 3px;
    text-align: center;
  }

  .forgot-pw-lock-icon {
    display: none;
  }

  .forgot-password-form .forgot-email-input {
    padding-left: 4px;
    padding-top: 5px;
  }

  .forgot-password-form {
    margin: 0.5rem;
  }
  input.send-new-pw-btn {
    height: 25px;
    /* background: #f2f2f2; */
  }
}

.character {
  image-rendering: pixelated;
}
