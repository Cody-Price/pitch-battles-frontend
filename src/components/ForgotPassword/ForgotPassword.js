import React, { Component } from "react";
import "./ForgotPassword.css";

import { forgotMyPasswordCall } from "../../utilities/fetchCalls";

class ForgotPassword extends Component {
  constructor() {
    super();
    this.state = {
      email: "",
      success: false,
      error: false,
      fetch: false
    };
  }

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  handleSubmit = async event => {
    event.preventDefault();
    this.setState({
      fetching: true
    });

    try {
      const response = await forgotMyPasswordCall(this.state.email);
      console.log(response);
      this.setState({
        success: true,
        fetching: false,
        error: false
      });
    } catch (error) {
      console.log(error.message);
      this.setState({ error: true, fetching: false, success: false });
    }
  };

  render() {
    return (
      <article className={`forgot-pw-wrapper ${this.props.status}`}>
        <p className="forgot-my-pw-label">forgot my password:</p>
        <form
          onSubmit={e => this.handleSubmit(e)}
          className="forgot-password-form"
        >
          <div className="password-flexbox">
            <p className="enter-email-for-new-pw">email</p>
            <div className="forgot-pw-lock-icon" />
            <input
              type="email"
              name="email"
              value={this.state.email}
              className="forgot-email-input"
              onChange={event => {
                this.handleChange(event);
              }}
            />
          </div>
          <p className={`pw-sent-success ${this.state.success}`}>
            new password sent - check your email
          </p>
          <p className={`pw-sent-failure ${this.state.error}`}>
            email does not match any account
          </p>

          <input
            type="submit"
            className="send-new-pw-btn"
            value="send new password"
          />
        </form>
        <button
          onClick={this.props.backToLogin}
          className="backup-from-forgot-pw"
        >
          login
        </button>
      </article>
    );
  }
}

export default ForgotPassword;
