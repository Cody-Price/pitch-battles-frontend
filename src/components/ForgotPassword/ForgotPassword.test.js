import React from "react";
import ForgotPassword from "./ForgotPassword";
import Enzyme, { shallow } from "enzyme";

const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("ForgotPassword", () => {
  let wrapper;
  let mockForgotPassword;
  let mockPreventDefault;
  let mockBackToLogin;

  beforeEach(() => {
    mockPreventDefault = jest.fn();
    mockForgotPassword = jest.fn();
    mockBackToLogin = jest.fn();
    wrapper = shallow(
      <ForgotPassword
        forgotPassword={mockForgotPassword}
        backToLogin={mockBackToLogin}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("handleChange", () => {
    it("should call handleChange on change", () => {
      const spy = jest.spyOn(wrapper.instance(), "handleChange");
      const mockEvent = {
        target: {
          name: "email",
          value: "kevin@simpson.com"
        }
      };

      wrapper.instance().forceUpdate();

      wrapper.find(".forgot-email-input").simulate("change", mockEvent);

      expect(spy).toHaveBeenCalledWith(mockEvent);
    });

    it("should setState", () => {
      const mockEvent = {
        target: {
          name: "email",
          value: "kevin@simpson.com"
        }
      };

      wrapper.instance().handleChange(mockEvent);

      expect(wrapper.state().email).toEqual(mockEvent.target.value);
    });
  });

  describe("handleSubmit", () => {
    it("should call handleSubmit on submit", () => {
      const mockEvent = {
        preventDefault: mockPreventDefault
      };
      const spy = jest.spyOn(wrapper.instance(), "handleSubmit");
      wrapper.instance().forceUpdate();

      wrapper.find(".forgot-password-form").simulate("submit", mockEvent);

      expect(spy).toHaveBeenCalled();
    });

    it("should call preventDefault", () => {
      const mockEvent = {
        preventDefault: mockPreventDefault
      };
      wrapper.instance().handleSubmit(mockEvent);

      expect(mockPreventDefault).toHaveBeenCalled();
    });

    it("should call forgotPasswordMyPasswordCall", () => {
      const expected = "myemail@mail.com";

      const mockEvent = {
        preventDefault: mockPreventDefault
      };
      wrapper.setState({
        email: expected
      });

      wrapper.instance().handleSubmit(mockEvent);

      // expect(mockForgotPassword).toHaveBeenCalledWith(expected);
    });
  });

  describe("backToLogin", () => {
    it("should call backToLogin on click", () => {
      wrapper.find(".backup-from-forgot-pw").simulate("click");

      expect(mockBackToLogin).toHaveBeenCalled();
    });
  });
});
