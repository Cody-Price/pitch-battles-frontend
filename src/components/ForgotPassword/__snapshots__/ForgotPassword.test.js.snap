// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword should match the snapshot 1`] = `
<article
  className="forgot-pw-wrapper undefined"
>
  <p
    className="forgot-my-pw-label"
  >
    forgot my password:
  </p>
  <form
    className="forgot-password-form"
    onSubmit={[Function]}
  >
    <div
      className="password-flexbox"
    >
      <p
        className="enter-email-for-new-pw"
      >
        email
      </p>
      <input
        className="forgot-email-input"
        name="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <input
      className="send-new-pw-btn"
      type="submit"
      value="send new password"
    />
  </form>
  <button
    className="backup-from-forgot-pw"
    onClick={[MockFunction]}
  >
    login
  </button>
</article>
`;
